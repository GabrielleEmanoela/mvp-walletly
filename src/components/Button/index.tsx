import React from 'react';
import { Button } from 'react-native-paper';
import { ViewStyle } from 'react-native';

import { styles } from './styles';

export type ButtonProps = {
  testID?: string;
  children: React.ReactNode | string;
  onPress: () => void;
  loading?: boolean;
  disabled?: boolean;
  contentStyle?: ViewStyle;
};

export const ButtonContainer = ({
  children,
  onPress,
  loading = false,
  disabled = false,
  contentStyle,
}: ButtonProps) => {
  return (
    <Button
      mode="contained"
      onPress={onPress}
      loading={loading}
      disabled={disabled}
      style={styles.button}
      labelStyle={styles.buttonText}
      contentStyle={contentStyle}
      testID={'ButtonContainer'}
    >
      {children}
    </Button>
  );
};
