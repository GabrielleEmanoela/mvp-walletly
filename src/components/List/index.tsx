import React from 'react';
import { View } from 'react-native';
import { FlashList } from '@shopify/flash-list';
import { Text, List, Avatar } from 'react-native-paper';
import { TransactionResponse } from '@/types/user';

import { styles } from './styles';

export type ListProps = {
  data: TransactionResponse[];
  onPress?: (item: { id: string }) => void;
  testID?: string;
};

export const ListComponent = ({ data, onPress }: ListProps) => {
  const handlePress = React.useCallback(
    (item: { id: string }) => () => onPress && onPress(item),
    [onPress],
  );

  return (
    <View style={styles.container} testID={'listComponent'}>
      {data?.length !== 0 ? (
        <FlashList
          data={data}
          estimatedItemSize={80}
          initialScrollIndex={0}
          keyExtractor={(item) => String(item.id)}
          contentContainerStyle={styles.contentContainer}
          renderItem={({ item }) => (
            <View testID={`item-${item.id}`}>
              <List.Item
                onPress={handlePress({ id: item.id })}
                title={item.description}
                description={
                  item.creditCard
                    ? `${item?.creditCard.brand} (${item?.creditCard.number})`
                    : ''
                }
                left={(props) => (
                  <Avatar.Icon
                    {...props}
                    size={40}
                    color={item.iconColor}
                    icon={item.icon || 'currency-usd'}
                    style={styles.icon}
                  />
                )}
                right={() => (
                  <Text style={styles.item}>
                    {item?.currency} {item?.amount}
                  </Text>
                )}
                style={styles.listItem}
              />
            </View>
          )}
          ItemSeparatorComponent={() => <View style={styles.separator} />}
        />
      ) : (
        <Text style={styles.emptyState}>Nenhuma transação encontrada.</Text>
      )}
    </View>
  );
};
