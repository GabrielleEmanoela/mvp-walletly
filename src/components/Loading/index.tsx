import React from 'react';
import { ActivityIndicator, View } from 'react-native';

import { styles } from './styles';

export const Loading = () => {
  return (
    <View style={styles.loadingContainer} testID="Loading">
      <ActivityIndicator animating size="large" />
    </View>
  );
};
