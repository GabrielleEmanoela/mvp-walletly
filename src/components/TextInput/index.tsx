import React from 'react';
import {
  TextInput as Input,
  TextInputProps as BaseProps,
  Text,
  TextInputIconProps,
} from 'react-native-paper';
import { View } from 'react-native';

import { errorStyles } from './styles';

export type TextInputProps = {
  onChangeText: (text: string) => void;
  errorMessage?: string;
  Icon?: React.ReactNode;
  testID?: string;
} & BaseProps;

export function TextInput({ errorMessage = '', ...props }: TextInputProps) {
  return (
    <>
      <Input {...props} left={props.Icon} testID={props.testID} />
      <View style={errorStyles.errorContainer}>
        <Text style={errorStyles.error}>{errorMessage}</Text>
      </View>
    </>
  );
}

export const TextInputIcon = ({ icon, ...rest }: TextInputIconProps) => {
  return <Input.Icon {...rest} icon={icon} testID={'TextInputIcon'} />;
};
