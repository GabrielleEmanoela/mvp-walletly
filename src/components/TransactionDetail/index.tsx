import React from 'react';
import { Avatar, Card, Text } from 'react-native-paper';

import { theme } from '@/theme/colors';
import { styles } from './styles';

export type TransactionDetailProps = {
  id: string;
  date: string;
  type: 'deposit' | 'withdrawal' | 'transfer' | 'payment';
  amount: number;
  currency: string;
  status: 'completed' | 'pending' | 'failed';
  description: string;
  fromAccount?: { name: string; accountNumber: string };
  toAccount?: { name: string; accountNumber: string };
  fee?: number;
  creditCard?: { number: string; brand: string };
  icon?: string;
  iconColor?: string;
};

function formatTitle(
  description: string,
  fromAccount?: { name: string; accountNumber: string },
) {
  if (fromAccount?.name) {
    return `${fromAccount.name}${fromAccount.accountNumber ? ` (${fromAccount.accountNumber})` : ''} - ${description}`;
  }
  return description;
}

function formatAmount(amount: number) {
  return amount.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

function formatDate(date: string) {
  return date ? new Date(date).toLocaleString('pt-BR') : '';
}

export const TransactionDetailCard = ({
  date,
  description,
  icon,
  iconColor,
  fromAccount,
  creditCard,
  amount,
  status,
  currency,
}: TransactionDetailProps) => (
  <Card style={styles.card} testID="transaction-detail-card">
    <Card.Title
      style={styles.title}
      title={formatTitle(description, fromAccount)}
      left={(props) =>
        icon ? (
          <Avatar.Icon
            {...props}
            icon={icon}
            style={[
              styles.icon,
              { backgroundColor: iconColor || theme.colors.primary },
            ]}
            testID="transaction-detail-icon"
          />
        ) : null
      }
    />
    <Card.Content>
      <Text style={styles.description} testID="transaction-detail-description">
        {description}
      </Text>
      {creditCard && (
        <Text style={styles.creditCard} testID="transaction-detail-creditcard">
          {creditCard.brand} ({creditCard.number})
        </Text>
      )}
      {date && (
        <Text style={styles.date} testID="transaction-detail-date">
          {formatDate(date)}
        </Text>
      )}
      <Text
        style={{
          ...styles.right,
          color:
            status === 'completed' ? theme.colors.success : theme.colors.danger,
        }}
        testID="transaction-detail-amount"
      >
        {formatAmount(amount)}
      </Text>
    </Card.Content>
  </Card>
);
