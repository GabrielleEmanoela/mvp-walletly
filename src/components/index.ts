export { Loading } from './Loading';
export { TextInputIcon, TextInput } from './TextInput';
