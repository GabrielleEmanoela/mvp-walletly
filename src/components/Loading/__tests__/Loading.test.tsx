import React from 'react';
import { render } from '@testing-library/react-native';
import { Loading } from '@/components';

const componentName = 'Loading';

describe(`${componentName}`, () => {
  it('should render ActivityIndicator', () => {
    const { getByTestId } = render(<Loading />);
    expect(getByTestId(componentName)).toBeOnTheScreen();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<Loading />);
    expect(toJSON()).toMatchSnapshot();
  });
});
