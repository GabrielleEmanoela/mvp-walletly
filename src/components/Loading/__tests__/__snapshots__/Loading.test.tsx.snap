// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="Loading"
>
  <ActivityIndicator
    animating={true}
    size="large"
  />
</View>
`;
